packages/stylefire/src/svg/utils.ts

```diff
@@ -0,0 +1,36 @@
+import { Dimensions } from './types';
+
+/**
+ * Feature detection for getBBox call on an unrendered SVG Element.
+ * Firefox will throw "NS_ERROR_FAILURE" error.
+ * We want to detect this upfront so we do not crash in any future measurement.
+ */
+export const CAN_MEASURE_NOT_RENDERED_SVG_ELEMENTS = (() => {
+  try {
+    document.createElementNS('http://www.w3.org/2000/svg', 'rect').getBBox();
+  } catch (e) {
+    return false;
+  }
+  return true;
+})();
+
+export const getDimensions = (
+  element: SVGElement | SVGPathElement
+): Dimensions =>
+  typeof (element as SVGGraphicsElement).getBBox === 'function'
+    ? (element as SVGGraphicsElement).getBBox()
+    : (element.getBoundingClientRect() as DOMRect);
+
+export const getSVGElementDimensionsSafely = (
+  element: SVGElement | SVGPathElement
+): Dimensions => {
+  if (CAN_MEASURE_NOT_RENDERED_SVG_ELEMENTS) {
+    return getDimensions(element);
+  }
+  try {
```

Comment:
is try/catch really that expensive nowadays? cant we just try/catch it always without calculating `CAN_MEASURE_NOT_RENDERED_SVG_ELEMENTS` upfront?

Reply:
According to Jason Miller (developit) its no longer an eager deopt and can be safely used (try/catch)

Reply:
@Andarist @InventingWithMonster should I simplify the PR to always use try/catch, or do we go with it as is?

Reply:
@niieani Yeah simplification is preferred, fewer potential sources of bugs.

Reply:
Switched to always try/catch.